### ISSUE TITLE
Fail to compile #c658f28

### ISSUE THREAD

**Issue by pcchen** [kaihsin/Cytnx#7]
Try to build

````
CC=/usr/local/bin/gcc-9 CXX=/usr/local/bin/g++-9 cmake ..
make
````

Fail with error message:
````
[ 98%] Building CXX object CMakeFiles/pycytnx.dir/pybind/cytnx.cpp.o
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp: In function 'void pybind11_init_cytnx(pybind11::module&)':
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: error: no matching function for call to 'pybind11::class_<cytnx_extension::CyTensor>::def(const char [11], <unresolved overloaded function type>)'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1084:13: note: candidate: 'template<class Func, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const char*, Func&&, const Extra& ...) [with Func = Func; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1084 |     class_ &def(const char *name_, Func&& f, const Extra&... extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1084:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   couldn't deduce template parameter 'Func'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1102:13: note: candidate: 'template<pybind11::detail::op_id id, pybind11::detail::op_type ot, class L, class R, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const pybind11::detail::op_<id, ot, L, R>&, const Extra& ...) [with pybind11::detail::op_id id = id; pybind11::detail::op_type ot = ot; L = L; R = R; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1102 |     class_ &def(const detail::op_<id, ot, L, R> &op, const Extra&... extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1102:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   mismatched types 'const pybind11::detail::op_<id, ot, L, R>' and 'const char [11]'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1114:13: note: candidate: 'template<class ... Args, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const pybind11::detail::initimpl::constructor<Args ...>&, const Extra& ...) [with Args = {Args ...}; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1114 |     class_ &def(const detail::initimpl::constructor<Args...> &init, const Extra&... extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1114:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   mismatched types 'const pybind11::detail::initimpl::constructor<Args ...>' and 'const char [11]'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1120:13: note: candidate: 'template<class ... Args, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(const pybind11::detail::initimpl::alias_constructor<Args ...>&, const Extra& ...) [with Args = {Args ...}; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1120 |     class_ &def(const detail::initimpl::alias_constructor<Args...> &init, const Extra&... extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1120:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   mismatched types 'const pybind11::detail::initimpl::alias_constructor<Args ...>' and 'const char [11]'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1126:13: note: candidate: 'template<class ... Args, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(pybind11::detail::initimpl::factory<Args ...>&&, const Extra& ...) [with Args = {Args ...}; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1126 |     class_ &def(detail::initimpl::factory<Args...> &&init, const Extra&... extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1126:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   mismatched types 'pybind11::detail::initimpl::factory<Args ...>' and 'const char [11]'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
      |                                                                         ^
In file included from /Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:4:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1132:13: note: candidate: 'template<class ... Args, class ... Extra> pybind11::class_<type_, options>& pybind11::class_<type_, options>::def(pybind11::detail::initimpl::pickle_factory<Args ...>&&, const Extra& ...) [with Args = {Args ...}; Extra = {Extra ...}; type_ = cytnx_extension::CyTensor; options = {}]'
 1132 |     class_ &def(detail::initimpl::pickle_factory<Args...> &&pf, const Extra &...extra) {
      |             ^~~
/Users/pcchen/gitlab/03132020/cytnx_kaishin/build/_deps/pybind11_sources-src/include/pybind11/pybind11.h:1132:13: note:   template argument deduction/substitution failed:
/Users/pcchen/gitlab/03132020/cytnx_kaishin/pybind/cytnx.cpp:933:73: note:   mismatched types 'pybind11::detail::initimpl::pickle_factory<Args ...>' and 'const char [11]'
  933 |                 .def("get_block_",&cytnx_extension::CyTensor::get_block_)
````

**Comment by kaihsin** [comment]
Hi, 
Thanks for raising the issue!!! 
Please try the latest push, as the issue might due to a temporal push.
Thanks a lot!!
Best,
Kai

**Comment by kaihsin** [comment]
Fixed in latest push 52d5082be343b0304ac77dd9b1a8a4791dac6d4e
